aws/elbv2.go

```diff
@@ -0,0 +1,53 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/elbv2"
+	"github.com/aws/aws-sdk-go/service/elbv2/elbv2iface"
+)
+
+func registerTargetsOnTargetGroups(svc elbv2iface.ELBV2API, targetGroupARNs []string, instances []string) error {
+	targets := make([]*elbv2.TargetDescription, len(instances))
+	for i, instance := range instances {
+		targets[i] = &elbv2.TargetDescription{
+			Id: aws.String(instance),
+		}
+	}
+
+	for _, targetGroupARN := range targetGroupARNs {
+		input := &elbv2.RegisterTargetsInput{
+			TargetGroupArn: aws.String(targetGroupARN),
+			Targets:        targets,
+		}
+
+		_, err := svc.RegisterTargets(input)
+		if err != nil {
+			return fmt.Errorf("unable to register instances %q in target group %s: %v", instances, targetGroupARN, err)
```

Comment:
Reading https://docs.aws.amazon.com/elasticloadbalancing/latest/APIReference/API_RegisterTargets.html errors, it seems that we should not stop in the loop.
I would suggest log an error and do a continue. Any thoughts on this behaviour?

Reply:
This error is logged and ignored in calling function (UpdateTargetGroupsAndAutoScalingGroups).

Reply:
But this would mean one ASG failure could block all the others, right?

Reply:
No. Look at UpdateTargetGroupsAndAutoScalingGroups function. It will log error and continue to next ASG. Error won't stop loop over autoScalingGroups.

Reply:
Ok thanks to clarify this